src/js/components/DSLInputField.js

```diff
@@ -72,7 +91,21 @@ class DSLInputField extends React.Component {
    * @param {SyntheticEvent} event - The change event
    */
   handleChange({target}) {
-    this.props.onChange(new DSLExpression(target.value));
+    this.setState({
+      expression: new DSLExpression(target.value)
+    });
+
+    clearTimeout(this.debounceTimer);
+    this.debounceTimer = setTimeout(
+      this.handleDebounceUpdate, DEBOUNCE_TIMEOUT
+    );
+  }
+
+  /**
+   * Forward property change update after the debounce timer expired
+   */
+  handleDebounceUpdate() {
```

Comment:
Is it make sense to have a `debounce` utility function and hide all the timers inside?

```Javascript
handleDebounceUpdate() {
  Util.debounce(
    () => this.props.onChange(this.state.expression),
    DEBOUNCE_TIMEOUT
  );
}
```